app/controllers/posts_controller.rb

```diff
@@ -1,6 +1,18 @@
 class PostsController < ApplicationController
+  PAGE_SIZE = 5
+
   def index
-    @posts = Post.all
+    if params[:sort] == 'top'
+      @posts = Post.all
+      descending = -1
+      @posts = @posts.sort_by do |post|
+        post.post_points * descending
+      end
```

Comment:
As we talked about earlier, I think this is a really good start! As part of your next work, a good exercise would be moving the entire sorting mechanism to the database.